File: src/kyu_6/reverse_or_rotate/js/solution.js
function revrot(str, sz) {
    if (sz <= 0 || !str || sz > str.length) return '';

    const sumCubes = chunk => chunk.split('').reduce((sum, digit) => sum += digit**3, 0);
    const reverse = chunk => chunk.split('').reverse().join('');
    const rotate = chunk => chunk.slice(1) + chunk.slice(0, 1);

    const chunks = [];

    for (let i = 0; i < str.length; i += sz) {
        let chunk = str.slice(i, i + sz);

        if (chunk.length < sz) continue;

        chunk = sumCubes(chunk) % 2
            ? rotate(chunk)
            : reverse(chunk);

        chunks.push(chunk)
    }

    return chunks.join('')
}

module.exports = revrot;

File: src/kyu_6/reverse_or_rotate/python/solution.py
def func(s):
    result = sum(int(x)**3 for x in s)
    if result % 2 == 0:
        return s[::-1]
    else:
        return s[1:] + s[0]

def rev_rot(s, sz):
    if not sz:
        return ''
    return ''.join(func(''.join(x)) for x in zip(*[iter(list(s))]*sz))